meshnesh/Bright-Events
# implement app mailing, email confirmation validation, reset password

# What does this PR do?

Implement sending emails to a user on registration, reset-password, email confirmation and rsvp

## Description of tasks to be completed

Implement methods to handle encoding and decoding confirmation tokens
define a function for sending out emails
Implement authentication token decorator
Improve code for token authentication
improve code on user rest email validation

## How should this be tested manually?

Register into the application with a POST request, you will get an email for registering.
To confirm an email send a POST request to `/api/auth/confirm`, you will receive an email with a url and a token.
Click the link and the email is confirmed through a PUT request to `/api/auth/verify/<token>`
To reset a password first we confirm the email of the user exists then send an email with the url to reset the password and the token to POST request `/api/auth/reset`
Click the link and set the new password which sends a PUT request to `/api/auth/reset-password/<token>`
Before an event is created the email of the user needs to have been confirmed, this is achieved by having a validation of the user email checking if it is validated.
